File: c-pr/45/c/code-quality-naming/app_config.c
#include <stdio.h>
#include <string.h>

typedef struct {
    char key[50];
    char value[100];
} ConfigEntry;

// {fact rule=code-quality-naming@v1.0 defects=1}
ConfigEntry cfg[100];
int cnt = 0;

void init() {
    cnt = 0;
}

void set(char* k, char* v) {
    strcpy(cfg[cnt].key, k);
    strcpy(cfg[cnt].value, v);
    cnt++;
}

char* get(char* k) {
    for (int i = 0; i < cnt; i++) {
        if (strcmp(cfg[i].key, k) == 0) {
            return cfg[i].value;
        }
    }
    return "";
}

int chk(char* k) {
    for (int i = 0; i < cnt; i++) {
        if (strcmp(cfg[i].key, k) == 0) {
            return 1;
        }
    }
    return 0;
}
// {/fact}

int main() {
    init();
    set("host", "localhost");
    printf("Host: %s\n", get("host"));
    printf("Has host: %d\n", chk("host"));
    return 0;
}

File: kotlin-pr/50/kotlin/code-quality-naming/AppConfig.kt
class AppConfig {
    // {fact rule=code-quality-naming@v1.0 defects=1}
    private var cfg: MutableMap<String, String>? = null

    fun init() {
        cfg = mutableMapOf()
    }

    fun set(k: String, v: String) {
        if (cfg == null) init()
        cfg!![k] = v
    }

    fun get(k: String): String {
        return cfg?.get(k) ?: ""
    }

    fun chk(k: String): Boolean {
        return cfg?.containsKey(k) ?: false
    }
    // {/fact}
}